From 3466658c918fb8bdcc7a6533fceeb8fb6f825f13 Mon Sep 17 00:00:00 2001
From: Sascha Wildner <saw@online.de>
Date: Thu, 21 Jul 2016 08:52:54 +0200
Subject: [PATCH 028/100] <sys/param.h>: Fix comments.

---
 sys/sys/param.h | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/sys/sys/param.h b/sys/sys/param.h
index d047e00..ae4d7fc 100644
--- a/sys/sys/param.h
+++ b/sys/sys/param.h
@@ -133,8 +133,8 @@
  * 400501 - unionfs removal
  * 400502 - private libraries: ssh ldns edit ncurses
  * 400503 - libarchive-3.0.2 import (add bsdcat)
- * 400400 - 4.6 release
- * 400500 - 4.7 development
+ * 400600 - 4.6 release
+ * 400700 - 4.7 development
  */
 #undef __DragonFly_version
 #define __DragonFly_version 400700	/* propagated to newvers */
-- 
2.7.2

