From 54a164037abb4ff14fdc4b9c8c4a5971b62df6d8 Mon Sep 17 00:00:00 2001
From: Sascha Wildner <saw@online.de>
Date: Sat, 23 Jul 2016 19:15:24 +0200
Subject: [PATCH 043/100] bsd-family-tree: Sync with FreeBSD.

---
 share/misc/bsd-family-tree | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/share/misc/bsd-family-tree b/share/misc/bsd-family-tree
index dee9b31..b4f429b 100644
--- a/share/misc/bsd-family-tree
+++ b/share/misc/bsd-family-tree
@@ -339,7 +339,11 @@ FreeBSD 5.2           |      |                 |                       |
  |  FreeBSD           |      |            OpenBSD 5.9                  |
  |   10.3             |      |                 |                       |
  |                    |      |                 |                       |
-FreeBSD 11 -current   |  NetBSD -current  OpenBSD -current     DragonFly -current
+ *--FreeBSD           |      |                 |                       |
+ |   11.0             |      |                 |                       |
+ |                    |      |                 |                       |
+ |                    |      |                 |                       |
+FreeBSD 12 -current   |  NetBSD -current  OpenBSD -current     DragonFly -current
  |                    |      |                 |                       |
  v                    v      v                 v                       v
 
@@ -751,4 +755,4 @@ Steven M. Schultz for providing 2.8BSD, 2.10BSD, 2.11BSD manual pages.
 Copyright (c) 1997-2012 Wolfram Schneider <wosch@FreeBSD.ORG>
 URL: http://svnweb.freebsd.org/base/head/share/misc/bsd-family-tree
 
-$FreeBSD: head/share/misc/bsd-family-tree 297551 2016-04-04 10:27:48Z maxim $
+$FreeBSD: head/share/misc/bsd-family-tree 303030 2016-07-19 16:55:16Z gjb $
-- 
2.7.2

