From 7c31553707fd7566f2a431481791a28c2f746e23 Mon Sep 17 00:00:00 2001
From: Sascha Wildner <saw@online.de>
Date: Thu, 28 Jul 2016 19:18:46 +0200
Subject: [PATCH 081/100] ed(1): Remove handling of non-POSIX environment.

---
 bin/ed/main.c | 15 +--------------
 1 file changed, 1 insertion(+), 14 deletions(-)

diff --git a/bin/ed/main.c b/bin/ed/main.c
index 669b9c1..d19606f 100644
--- a/bin/ed/main.c
+++ b/bin/ed/main.c
@@ -58,11 +58,7 @@
 #include "ed.h"
 
 
-#ifdef _POSIX_SOURCE
 static sigjmp_buf env;
-#else
-static jmp_buf env;
-#endif
 
 /* static buffers */
 char stdinbuf[1];		/* stdin buffer */
@@ -144,12 +140,7 @@ top:
 	signal(SIGHUP, signal_hup);
 	signal(SIGQUIT, SIG_IGN);
 	signal(SIGINT, signal_int);
-#ifdef _POSIX_SOURCE
-	if ((status = sigsetjmp(env, 1)))
-#else
-	if ((status = setjmp(env)))
-#endif
-	{
+	if ((status = sigsetjmp(env, 1))) {
 		fputs("\n?\n", stderr);
 		errmsg = "interrupt";
 	} else {
@@ -1372,11 +1363,7 @@ handle_int(int signo)
 	if (!sigactive)
 		quit(1);
 	sigflags &= ~(1 << (signo - 1));
-#ifdef _POSIX_SOURCE
 	siglongjmp(env, -1);
-#else
-	longjmp(env, -1);
-#endif
 }
 
 
-- 
2.7.2

